openedx/core/djangoapps/safe_sessions/middleware.py

```diff
@@ -715,11 +785,15 @@ def __setattr__(self, name, value):
                 location = "\n".join("%30s : %s:%d" % (t[3], t[1], t[2]) for t in stack[0:12])
 
                 if not hasattr(self, 'debug_user_changes'):
+                    # list of string debugging info for each user change (e.g. user id, stack trace, etc.)
                     self.debug_user_changes = []  # pylint: disable=attribute-defined-outside-init
+                    # list of changed user ids
+                    self.user_id_list = []  # pylint: disable=attribute-defined-outside-init
 
                 if not hasattr(request, name):
                     original_user = value
                     if hasattr(value, 'id'):
+                        self.user_id_list.append(value.id)
```

Comment:
**Reviewer Note:** I didn't want to think too hard about this, so I decided to just hook into the existing/working logging code to grab the ids. Please mark resolved after reading.